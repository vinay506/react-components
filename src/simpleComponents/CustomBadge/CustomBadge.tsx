import React from 'react'
import styled from 'styled-components';
function CustomBadge(props: any) {
    /**
     * Text: Content to be displayed
     * bgColor: Backhround color of a badge
     * color: color of the text
     * position: Use when we want to display the batch over any container. Make sure make your container position as relative
     * (top, right, left, bottom): If position is absolute then use these properties to display the bacth over any container at any position
     * borderRadius: If required but default is 5px
     */
    const { text, bgColor, color, position, top, right, left, bottom, borderRadius } = props;
    const badgeInlineStyle = {
        backgroundColor: bgColor,
        color: color,
        position: position,
        top: top,
        right: right,
        left: left,
        bottom: bottom,
        borderRadius: borderRadius
    };
    return (
        <React.Fragment>
            <BadgeSpan id="uniqueBadge" style={badgeInlineStyle}>
                {text}
            </BadgeSpan>
        </React.Fragment>
    )
}

const BadgeSpan = styled.span`
    background-color: blue;
    color: white;
    padding: 5px;
    border-radius: 5px;
    height: 10px;
    min-width: 20px;
    font-size: 12px;
`

export default CustomBadge
