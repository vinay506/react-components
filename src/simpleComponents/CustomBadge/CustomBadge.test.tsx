import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import CustomBadge from './CustomBadge';
import 'jest-styled-components';

describe('Custom Button Outlined', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any;
    beforeEach(() => {
        propInfo = {
            text: 'Recomended',
            bgColor: 'gray',
            color: 'white',
            position: 'absolute',
            top: '20px',
            right: '20px',
            left: '20px',
            bottom: '10px',
            borderRadius: '10px'
        }

        defaultStyles = {
            backgroundColor: 'blue',
            color: 'white',
            padding: '5px',
            borderRadius: '5px',
            height: '10px',
            minWidth: '20px',
            fontSize: '12px'
        }

        wrapper = shallow(<CustomBadge {...propInfo} />);
    });


    test('Custom Badge Snapshot', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Custom Badge Element check', () => {
        const elementLength = wrapper.find('#uniqueBadge').length;
        expect(elementLength).toEqual(1);
    })
    // Props Validation Check
    test('Custom Badge Color Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge').text();
        expect(elementStyles).toEqual(propInfo.text);
    })
    test('Custom Badge Color Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.color).toEqual(propInfo.color);
    })
    test('Custom Badge Background Color Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.backgroundColor).toEqual(propInfo.bgColor);
    })
    test('Custom Badge Position Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.position).toEqual(propInfo.position);
    })
    test('Custom Badge Top Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.top).toEqual(propInfo.top);
    })

    test('Custom Badge Bottom Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.bottom).toEqual(propInfo.bottom);
    })

    test('Custom Badge Right Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.right).toEqual(propInfo.right);
    })

    test('Custom Badge Left Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.left).toEqual(propInfo.left);
    })

    test('Custom Badge Left Validation', () => {
        const elementStyles = wrapper.find('#uniqueBadge');
        expect(elementStyles.props().style.borderRadius).toEqual(propInfo.borderRadius);
    })

    // Defualt Style Validation
    test('Custom Badge Default Background Color Validation', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.backgroundColor);
    })

    test('Custom Badge Default Color Validation', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.color);
    })

    test('Custom Badge Default Padding Validation', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.padding);
    })

    test('Custom Badge Default Border Radius Validation', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.borderRadius);
    })
    test('Custom Badge Default Height Validation', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.height);
    })

    test('Custom Badge Default Min Width Validation', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.minWidth);
    })

    test('Custom Badge Default Font Size Validation', () => {
        const instance = renderer.create(<CustomBadge {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontSize);
    })


})