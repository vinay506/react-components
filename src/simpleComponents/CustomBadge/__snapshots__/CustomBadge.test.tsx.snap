// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button Outlined Custom Badge Snapshot 1`] = `
.c0 {
  background-color: blue;
  color: white;
  padding: 5px;
  border-radius: 5px;
  height: 10px;
  min-width: 20px;
  font-size: 12px;
}

<span
  className="c0"
  id="uniqueBadge"
  style={
    Object {
      "backgroundColor": "gray",
      "borderRadius": "10px",
      "bottom": "10px",
      "color": "white",
      "left": "20px",
      "position": "absolute",
      "right": "20px",
      "top": "20px",
    }
  }
>
  Recomended
</span>
`;
